flang/test/Lower/array-constructor-2.f90

```diff
@@ -104,8 +104,8 @@ subroutine test4(a, b, n1, m1)
   integer, external :: f1, f2, f3
 
   !  Dynamic array ctor with dynamic extent using implied do loops.
-  ! CHECK-DAG: fir.alloca index {uniq_name = ".buff.pos"}
-  ! CHECK-DAG: fir.alloca index {uniq_name = ".buff.size"}
+  ! CHECK-DAG: fir.alloca index {pinned, uniq_name = ".buff.pos"}
+  ! CHECK-DAG: fir.alloca index {pinned, uniq_name = ".buff.size"}
```

Comment:
I'm confused why these compiler generated temps are now pinned. In fact, that seems incorrect. For example, if these were in a subprogram that was inlined into another subprogram, I'd expect these to be hoisted to the entry of the calling subprogram, etc.

Reply:
OK. I did not consider inlining. I will remove pinned for Fortran semantics.